container/root/worker.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Entrypoint for utilizing as a worker pool instead of a web server
+# Based on configuration, can run multiple instances of a single worker process
+
+# TODO: this is straight copy-paste from docker-nginx's run.sh, needs to be split into a separate file than the nginx-start
+
+RUN_SCRIPTS=/run.d
+STATUS=0
+SUPERVISOR_CONF=/etc/supervisor/conf.d/worker.conf
+
+export CONTAINER_ROLE='worker'
+
+
+# Run shell scripts (ending in .sh) in run.d directory
+
+# When .sh run scripts fail (exit non-zero), container run will fail
+# NOTE: if a .sh script exits with 99, this is our stop signal, container will exit cleanly
+
+for file in $RUN_SCRIPTS/*.sh; do
+
+  echo "[run.d] executing ${file}"
+
+  /bin/bash $file
+
+  STATUS=$?  # Captures exit code from script that was run
+
+  if [[ $STATUS == $SIGNAL_BUILD_STOP ]]
+  then
+    echo "[run.d] exit signalled - ${file}"
+    exit # Exit cleanly
+  fi
+
+  if [[ $STATUS != 0 ]]
+  then
+    echo "[run.d] failed executing - ${file}"
+    exit $STATUS
+  fi
+
+done
+
+
+WORKER_RUNNER=$1
+WORKER_QUANTITY=$2
+WORKER_NAME=$3
+WORKER_PARAM_1=$4
+WORKER_PARAM_2=$5
+
+echo "[worker] Runner: ${WORKER_RUNNER} Name: ${WORKER_NAME} Quantity ${WORKER_QUANTITY}"
+
+echo "\
+[program:worker]
+command=${WORKER_RUNNER} ${WORKER_PARAM_1} ${WORKER_PARAM_2}
+process_name=%(program_name)s%(process_num)s
+numprocs=${WORKER_QUANTITY}
+autorestart=true
+redirect_stderr=true
+stdout_logfile=/dev/stdout" > $SUPERVISOR_CONF
```

Comment:
That's a really good point.  However, might that result in errant/invalid/duplicate config lines in cases where the container is tagged and used repeatedly (e.g., `docker-compose`)? I'm not sure if this will be used in this mode within local development environments, but I don't see why not.  Either way, moving this as-is to above the run.d loop would solve that, so long as the other scripts used `>>`.  An alternative to guarantee no overwrite could be to reset the file atop this script instead and use the append here.